### ISSUE TITLE
authserver_ldap module bugs (transport/urltype handling, CA requirement)

### ISSUE THREAD

**Issue by darksoul42** [pfsensible/core#3]
I tried setting LDAP authentication via the module, only to find the following quirks :
- Even though I am requesting a TCP transport, `state=present` forces it to require a CA parameter
-- Removing the requirement from the `required_if` definition works
- Whether transport is TCP or not is checked against `module.params['transport']`, but it is not defined and using ldap_urltype is not appropriate
-- Adding the following line `authserver['transport'] = module.params['transport']` fixes the problem

**Comment by darksoul42** [comment]
Proposed diff :
```diff
--- pfsensible/core/plugins/modules/authserver_ldap.py.old	2020-01-23 18:42:27.059170029 +0900
+++ pfsensible/core/plugins/modules/authserver_ldap.py	2020-01-23 18:43:44.747709687 +0900
@@ -208,7 +208,7 @@
             'attr_groupobj': {'default': 'posixGroup', 'type': 'str'},
         },
         required_if=[
-            ["state", "present", ["host", "port", "transport", "ca", "scope"]],
+            ["state", "present", ["host", "port", "transport", "scope"]],
             ["transport", "starttls", ["ca"]],
             ["transport", "ssl", ["ca"]],
         ],
@@ -226,6 +226,7 @@
         authserver['host'] = module.params['host']
         authserver['ldap_port'] = module.params['port']
         urltype = dict({'tcp': '', 'starttls': '', 'ssl': 'SSL - Encrypted'})
+        authserver['transport'] = module.params['transport']
         authserver['ldap_urltype'] = urltype[module.params['transport']]
         authserver['ldap_protver'] = module.params['protver']
         authserver['ldap_timeout'] = module.params['timeout']
```

**Comment by opoplawski** [comment]
Thanks for the report.  This should be fixed now along with some other enhancements/fixes with commit 82724c3
